internal/core/entities/scheduler.go

```diff
@@ -43,12 +44,28 @@ const (
 )
 
 type Scheduler struct {
-	Name            string `validate:"min=1"`
-	Game            string `validate:"min=1"`
-	State           string
+	Name            string `validate:"required"`
+	Game            string `validate:"required"`
+	State           string `validate:"required"`
 	RollbackVersion string
 	Spec            game_room.Spec
 	PortRange       *PortRange
 	CreatedAt       time.Time
-	MaxSurge        string `validate:"min=1"`
+	MaxSurge        string `validate:"required"`
+}
+
+func NewScheduler(name string, game string, state string, maxSurge string, spec game_room.Spec, portRange *PortRange) (*Scheduler, error) {
+	scheduler := &Scheduler{
+		Name:      name,
+		Game:      game,
+		State:     state,
+		Spec:      spec,
+		PortRange: portRange,
+		MaxSurge:  maxSurge}
+	return scheduler, scheduler.Validate()
+}
+
+func (s *Scheduler) Validate() error {
+	err := validations.Validate.Struct(s)
```

Comment:
@lyracampos when we run this validate method, does it validates inner structs also? like PortRange and Spec?

Reply:
@GuilhermBrSp yes it does! all of structs will be checked when scheduler validate is call